### PR TITLE
Create document highlighting dev and deploy

### PR DESCRIPTION
**Pull request by cdrage**
Create document highlighting dev and deploy

**What type of PR is this:**

/kind documentation

**What does this PR do / why we need it:**

* Changes title of "Basics" to "Concepts" in concepts doc
* Adds document on how to do dev vs deploy and which command you should
  use
* Highlights when to use either command

**Which issue(s) this PR fixes:**

Closes https://github.com/redhat-developer/odo/issues/6083

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
When reviewing, please add any suggestions you have!

**Comment by cdrage** [comment]
Ready for another review!

**Comment by kadel** [comment]
I think this is a good start. It addresses the most frequent confusion that I say.

/approve

File: docs/website/docs/overview/dev_and_deploy.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## When should I use `odo dev`?
+
+`odo dev` should be used in the initial development process of your application. 
+
+For example, you should use `odo dev` when you are:
+* Making changes constantly
+* Want to preview any changes
+* Testing initial Kubernetes support for your application
+* Want to debug and run tests
+* Deploy privately on a local development environment
```
Comment (valaparthvi) [inline docs/website/docs/overview/dev_and_deploy.md:21]:
```suggestion
* Run the application on a local development environment
```
Or something with which we can avoid using "Deploy".

File: docs/website/docs/overview/dev_and_deploy.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+## When should I use `odo dev`?
+
+`odo dev` should be used in the initial development process of your application. 
+
+For example, you should use `odo dev` when you are:
+* Making changes constantly
+* Want to preview any changes
+* Testing initial Kubernetes support for your application
+* Want to debug and run tests
+* Run the application on a local development environment
```
Comment (rm3l) [inline docs/website/docs/overview/dev_and_deploy.md:21]: A small comment about the tense of the verbs, which I found hard to read in conjunction with "... when you are ".

I would suggest something like:
```
For example, you should use `odo dev` when:
* you are making changes constantly
* you want to preview any changes
* you are testing initial Kubernetes support for your application
* you want to debug and run tests
* you want to run the application on a local development environment
```

What do you think?

File: docs/website/docs/overview/dev_and_deploy.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+## When should I use `odo deploy`?
+
+`odo deploy` should be the deploy stage of development when you are ready for a "production ready" environment.
+
+For example, you should use `odo deploy` when you are:
+* Ready for the application to be viewed publically
+* Require building and pushing the container
+* Needing custom Kubernetes YAML for your production environment
```
Comment (rm3l) [inline docs/website/docs/overview/dev_and_deploy.md:30]: Same point as before.

**Comment by valaparthvi** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests

File: docs/website/docs/overview/dev_and_deploy.md (hunk 2 above)
Comment (cdrage) [reply docs/website/docs/overview/dev_and_deploy.md:21]: IMO want to avoid repeating the same sentence twice starting with the same word

Comment (valaparthvi) [reply docs/website/docs/overview/dev_and_deploy.md:21]: I agree with Charlie on this one. It's much nicer to read with the current change, and it also makes more sense.

File: docs/website/docs/overview/dev_and_deploy.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+The two most important commands in `odo` are `odo dev` and `odo deploy`. 
+
+In some situations, you'd want to use [`odo dev`](/docs/command-reference/dev) over [`odo deploy`](/docs/command-reference/deploy) and vice-versa. This document highlights when you should use either command.
+
+## When should I use `odo dev`?
+
+`odo dev` should be used in the initial development process of your application. 
+
+For example, you should use `odo dev` when you are:
```
Comment (valaparthvi) [inline docs/website/docs/overview/dev_and_deploy.md:16]:
```suggestion
For example, you should use `odo dev` when you are working with a local development environment and:
```

File: docs/website/docs/overview/dev_and_deploy.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+* Want to preview any changes
+* Testing initial Kubernetes support for your application
+* Want to debug and run tests
+* Run the application on a local development environment
+
+## When should I use `odo deploy`?
+
+`odo deploy` should be the deploy stage of development when you are ready for a "production ready" environment.
+
+For example, you should use `odo deploy` when you are:
```
Comment (valaparthvi) [inline docs/website/docs/overview/dev_and_deploy.md:27]:
```suggestion
For example, you should use `odo deploy` when you are working with a production environment and:
```

File: docs/website/docs/overview/dev_and_deploy.md (hunk 2 above)
Comment (rm3l) [reply docs/website/docs/overview/dev_and_deploy.md:21]: Alright - maybe it's just me then :D

Comment (rm3l) [reply docs/website/docs/overview/dev_and_deploy.md:21]: I found the previous change hard to read at the time I reviewed this PR a few days back:

```
For example, you should use odo dev when you are:
- making changes constantly
- want to preview any changes
- ...
```

But it looks better now with the latest changes you suggested: 
```
For example, you should use odo dev when you are working with a local development environment and:
- making changes constantly
- want to preview any changes
- ...
```

File: docs/website/docs/overview/dev_and_deploy.md (hunk 4 above)
Comment (rm3l) [reply docs/website/docs/overview/dev_and_deploy.md]: The development environment is not necessarily local, no? I think we should just mention a "development environment".

File: docs/website/docs/overview/dev_and_deploy.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+## When should I use `odo dev`?
+
+`odo dev` should be used in the initial development process of your application. 
+
+For example, you should use `odo dev` when you are working with a local development environment and:
+* Making changes constantly
+* Want to preview any changes
+* Testing initial Kubernetes support for your application
+* Want to debug and run tests
+* Run the application on a local development environment
```
Comment (valaparthvi) [inline docs/website/docs/overview/dev_and_deploy.md:21]: Can we make the sentences lower-cased since they are in continuation? Same for `odo deploy`.

**Review by valaparthvi — CHANGES_REQUESTED** [review]

**Comment by cdrage** [comment]
Good suggestions! Thanks @valaparthvi and @rm3l 

I've gone ahead and switched the sentences to be more grammatically correct and in lower case. Thanks!

**Review by rm3l — APPROVED** [review]

**Comment by cdrage** [comment]
I rebased, so re-adding lgtm label

**Comment by cdrage** [comment]
/override Kubernetes-Integration-Tests/Kubernetes-Integration-Tests

**Pull request merged**